src/text/text_data_source.rs

```diff
@@ -25,7 +25,7 @@ impl<'a> TextIonDataSource for &'a str {
     }
 }
 
-impl<'a> TextIonDataSource for &'a &[u8] {
+impl<'a> TextIonDataSource for &'a [u8] {
```

Comment:
PR tour: the second `&` was a typo that I didn't notice until the compiler brought my attention to the crazy lifetimes it was introducing.